beginner_source/blitz/tensor_tutorial.py

```diff
@@ -1,195 +1,166 @@
-# Construct a tensor directly from data:
-
-x = torch.tensor([5.5, 3])
-print(x)
 
 ###############################################################
-# or create a tensor based on an existing tensor. These methods
-# will reuse properties of the input tensor, e.g. dtype, unless
-# new values are provided by user
-
-x = x.new_ones(5, 3, dtype=torch.double)      # new_* methods take in sizes
-print(x)
+# **From another tensor:**
+# The new tensor retains the properties of the arg tensor, unless explicitly overridden
 
-x = torch.randn_like(x, dtype=torch.float)    # override dtype!
-print(x)                                      # result has the same size
+new_ones_tnsr = torch.ones_like(tnsr_from_data) # 2 x 2 matrix of ones
 
-###############################################################
-# Get its size:
+try:
+  new_rand_tnsr = torch.rand_like(tnsr_from_data) # 2 x 2 matrix of random numbers
+except RuntimeError as e:
+  print(f"RuntimeError thrown: {e}")
+  print()
+  print(f"Random values in PyTorch are floating points. Datatype passed to torch.rand_like(): {tnsr_from_data.dtype}")
 
-print(x.size())
 
-###############################################################
-# .. note::
-#     ``torch.Size`` is in fact a tuple, so it supports all tuple operations.
-#
-# Operations
-# ^^^^^^^^^^
-# There are multiple syntaxes for operations. In the following
-# example, we will take a look at the addition operation.
+######################################################################
+# This works after we override the dtype property
 #
-# Addition: syntax 1
-y = torch.rand(5, 3)
-print(x + y)
 
-###############################################################
-# Addition: syntax 2
+new_rand_tnsr = torch.rand_like(tnsr_from_data, dtype=torch.float)
+print(new_rand_tnsr)
 
-print(torch.add(x, y))
 
-###############################################################
-# Addition: providing an output tensor as argument
-result = torch.empty(5, 3)
-torch.add(x, y, out=result)
-print(result)
 
-###############################################################
-# Addition: in-place
+######################################################################
+# --------------
+#
 
-# adds x to y
-y.add_(x)
-print(y)
 
-###############################################################
-# .. note::
-#     Any operation that mutates a tensor in-place is post-fixed with an ``_``.
-#     For example: ``x.copy_(y)``, ``x.t_()``, will change ``x``.
+######################################################################
+# Tensor Attributes
+# ~~~~~~~~~~~~~~~~~
+#
+# Tensors have attributes that describe their contents and functions. They
+# also have attributes that autograd uses to keep track of them in the
+# computational graph (more on this in the next section).
+#
+# **Docs Issues** - https://pytorch.org/docs/stable/tensor_attributes.html
+# is not comprehensive (missing data, grad, grad_fn, shape). Contains
```

Comment:
`data` and `grad_fn` are explicitly not mentioned as they should not be used. They are just left for backward compatibility. And will hopefully go away soon(ish).
`grad` and `shape` do appear to be missing.